src/engine/client/glshader.cpp

```diff
@@ -53,9 +53,16 @@ GLuint LoadShader(const char *vertex_path, const char *fragment_path) {
     // Check vertex shader
     glGetShaderiv(vertShader, GL_COMPILE_STATUS, &result);
     glGetShaderiv(vertShader, GL_INFO_LOG_LENGTH, &logLength);
-    //std::vector vertShaderError((logLength > 1) ? logLength : 1);
-    //glGetShaderInfoLog(vertShader, logLength, NULL, &vertShaderError[0]);
-    //std::cout << &vertShaderError[0] << std::endl;
+    if (!result)
+    {
+        dbg_msg("gfx", "Could not compile vertex shader %s", vertex_path);
+        char *infoLog = (char *) malloc ( sizeof ( char ) * logLength );
+        glGetShaderInfoLog(vertShader, logLength, NULL, infoLog);
+        dbg_msg("gfx", "%s", infoLog);
+        free ( infoLog );
+        dbg_msg("gfx", "%s", vertShaderSrc);
```

Comment:
Seriously, I would better not spam the console with the whole shader source code here, it would push the log out of view... Since you above print the file path where the error occurred, one who wants to fix it can then open this file himself to have a look at the code.